sycl/test-e2e/ESIMD/unified_memory_api/Inputs/scatter.hpp

```diff
@@ -486,19 +486,16 @@ bool testLACC(queue Q, uint32_t MaskStride,
          uint32_t GlobalElemOffset = GlobalID * N;
          uint32_t LocalElemOffset = LocalID * N;
 
-         if (LocalID == 0) {
-           for (int I = 0; I < Threads * N; I++) {
-             simd<T, 1> InVec(Out + GlobalElemOffset + I);
-             LocalAcc[I] = InVec[0];
-           }
+         for (int I = 0; I < Size; I++) {
+           LocalAcc[I] = Out[I];
          }
-         barrier();
 
-         simd<uint32_t, NOffsets> ByteOffsets(LocalElemOffset * sizeof(T),
+         simd<uint32_t, NOffsets> ByteOffsets(GlobalElemOffset * sizeof(T),
```

Comment:
Sorry, but this test "Refactor the test" commit does not seem good. Perhaps, it is some temporary fix/experiment.

The main idea of using SLM is to reserve the shared memory for 1 work-group. This commit though reserves too much SLM which is enough for all 8 work-groups and have writes/reads to indexes unique for different work-groups, which is not good not only because it is too much SLM, but it significantly reduces the test strength. 
For example, if accidentally (or due to implementational error) slm_gather/scatter starts reading/writing from/to some random global memory, you would never know about that unless such writes erase something important in global memory. With smaller SLM it would be possible to catch it.